code/datums/mutations/holy_mutation/honorbound.dm

```diff
@@ -0,0 +1,125 @@
+	RegisterSignal(owner, COMSIG_ATOM_HULK_ATTACK, .proc/hulk_guilt)
+	RegisterSignal(owner, COMSIG_ATOM_ATTACK_HAND, .proc/hand_guilt)
+	RegisterSignal(owner, COMSIG_ATOM_ATTACK_PAW, .proc/paw_guilt)
+	RegisterSignal(owner, COMSIG_ATOM_BULLET_ACT, .proc/bullet_guilt)
+	RegisterSignal(owner, COMSIG_ATOM_HITBY, .proc/thrown_guilt)
+
+	//signal that checks for dishonorable attacks
+	RegisterSignal(owner, COMSIG_MOB_CLICKON, .proc/attack_honor)
+
+/datum/mutation/human/honorbound/on_losing(mob/living/carbon/human/owner)
+	UnregisterSignal(owner, list(COMSIG_PARENT_ATTACKBY,COMSIG_ATOM_HULK_ATTACK,COMSIG_ATOM_ATTACK_HAND,COMSIG_ATOM_ATTACK_PAW,COMSIG_ATOM_BULLET_ACT,COMSIG_ATOM_HITBY,COMSIG_MOB_CLICKON))
+	. = ..()
+
+/datum/mutation/human/honorbound/proc/attack_honor(mob/living/carbon/human/honorbound, atom/clickingon, params)
+	SIGNAL_HANDLER
+	var/obj/item/weapon = honorbound.get_active_held_item()
+	var/list/modifiers = params2list(params)
+
+	if(!isliving(clickingon))
+		return
+	//fucking YUGE check but essentially is every situation where a click shouldn't be checked for honor:
+	//non combat mode clicks:
+	//* no weapon, not disarming, weapon has no force
+	//combat mode clicks
+	//* is not a gun and out of range of attacking
+	if(!honorbound.DirectAccess(clickingon) && !isgun(weapon))
+		return
+	if(!honorbound.combat_mode && ((!weapon || !weapon.force) && !LAZYACCESS(modifiers, RIGHT_CLICK)))
+		return
+	var/mob/living/clickedmob = clickingon
+	if(!is_honorable(honorbound, clickedmob))
+		return COMSIG_MOB_CANCEL_CLICKON
+
+/datum/mutation/human/honorbound/proc/guilty(mob/living/user)
+	if(user in guilty)
+		return
+	var/datum/mind/guilty_conscience = user.mind
+	if(guilty_conscience) //sec and medical are immune to becoming guilty through attack (we don't check holy because holy shouldn't be able to attack eachother anyways)
+		var/job = guilty_conscience.assigned_role
+		if(job in (GLOB.security_positions + GLOB.medical_positions))
+			return
+	to_chat(owner, "<span class='notice'>[user] is now considered guilty by [GLOB.deity] for attacking you first.</span>")
+	guilty += user
+
+/datum/mutation/human/honorbound/proc/is_honorable(mob/living/carbon/human/honorbound_human, mob/living/target_creature)
+	var/is_guilty = (target_creature in guilty)
+	//THE UNREADY (Applies over ANYTHING else!)
+	if(target_creature.IsSleeping() || HAS_TRAIT(target_creature, TRAIT_RESTRAINED))
+		to_chat(honorbound_human, "<span class='warning'>There is no honor in attacking the <b>unready</b>.</span>")
+		return FALSE
+	//THE JUST (Applies over guilt except for med, so you best be careful!)
+	if(ishuman(target_creature))
+		var/mob/living/carbon/human/target_human = target_creature
+		var/job = target_human.mind?.assigned_role
+		var/is_holy = target_human.mind?.holy_role
+		if(job in GLOB.security_positions || is_holy)
+			to_chat(honorbound_human, "<span class='warning'>There is nothing rightous in attacking the <b>just</b>.</span>")
+			return FALSE
```

Comment:
this can be used to detect if someone has stolen the identity of a sec officer or actually is one

maybe make this a mindshield check instead? idk

Reply:
The solution of changing it to windshield makes a lot more issues than it solves, you'll have to take my word on this one